streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredTimestampedKeyValueStore.java

```diff
@@ -205,12 +206,12 @@ private <R> QueryResult<R> runTimestampedRangeQuery(final Query<R> query,
         final QueryResult<R> result;
         final TimestampedRangeQuery<K, V> typedQuery = (TimestampedRangeQuery<K, V>) query;
         RangeQuery<Bytes, byte[]> rawRangeQuery;
-        final boolean isKeyAscending = typedQuery.isKeyAscending();
+        ResultOrder order = typedQuery.resultOrder();
         rawRangeQuery = RangeQuery.withRange(
                 keyBytes(typedQuery.lowerBound().orElse(null)),
                 keyBytes(typedQuery.upperBound().orElse(null))
         );
-        if (!isKeyAscending) {
+        if (order.equals(ResultOrder.DESCENDING)) {
```

Comment:
We need to handle the ascending case, too

Reply:
https://github.com/apache/kafka/pull/14907#discussion_r1416481443